src/main/java/org/jenkinsci/plugins/sshsteps/util/SSHStepExecution.java

```diff
@@ -63,18 +66,20 @@ static synchronized ExecutorService getExecutorService() {
   public final boolean start() {
     Authentication auth = Jenkins.getAuthentication();
     task = getExecutorService().submit(() -> {
+      threadName = Thread.currentThread().getName();
       try {
-        getContext()
-            .onSuccess(ACL.impersonate(auth, new NotReallyRoleSensitiveCallable<T, Exception>() {
-              @Override
-              public T call() throws Exception {
-                threadName = Thread.currentThread().getName();
-                MDC.put("execution.id", UUID.randomUUID().toString());
-                return SSHStepExecution.this.run();
-              }
-            }));
-      } catch (Exception e) {
-        getContext().onFailure(e);
+        MDC.put("execution.id", UUID.randomUUID().toString());
+        T ret;
+        try (ACLContext acl = ACL.as(auth)) {
+          ret = run();
+        }
+        getContext().onSuccess(ret);
+      } catch (Throwable x) {
```

Comment:
Catching Throwable is bad practice. Is this necessary?

Reply:
I have used the latest code from dependent plugin, so I would keep as it is. 

https://github.com/jenkinsci/workflow-step-api-plugin/blob/master/src/main/java/org/jenkinsci/plugins/workflow/steps/SynchronousNonBlockingStepExecution.java#L50-L56